### ISSUE TITLE
This project doesn't seem like an operating system

### ISSUE THREAD

**Issue by sharpbai** [fiatrete/OpenDAN-Personal-AI-OS#23]
Although the project's name includes "OS" 
I haven't seen any code related to an operating system.

Is it really an operating system project?

**Comment by DiligentCatCat** [comment]
Indeed, it's only a prototype, nothing about `OS` yet.

The reason that we release this very start Jarvis, is that want people use Jarvis to feel what an AIOS might be in the future, to feel that how an AI would be used by people.

As for the `OS`, we think it should be an environment that `AI`s can be integrated easily, and `AI`s can interact with each other in a uniform and easy way. Thus developing and distributing AI based applications would be easy.

We are thinking the formal architecture recently.

**Comment by p6002** [comment]
I just installed as VM and it looks like normal Ubuntu. What's next?

**Comment by fiatrete** [comment]
> I just installed as VM and it looks like normal Ubuntu. What's next?

You are right, right the os_installer is merely a placeholder, signifying our ultimate goal of building an AI-based operating system. The current code in the repository is for our butler agent, Jarvis. We are in the process of designing and prototyping the code for the OS. We estimate that it will be available within a month. Please stay tuned for updates.

**Comment by p6002** [comment]
I'll follow up, but the current readme on Github, along with the description, images and videos, suggests that at least what's in the images is available. 
So far the iso image has been downloading/creating for an hour, I installed this on a VM in Proxmox, and there is normal Ubuntu :) no IP addresses, ports from the graphics in the Readme work. That's why people are disappointed with this "marketing campaign" in recent days.

**Comment by fiatrete** [comment]
> I'll follow up, but the current readme on Github, along with the description, images and videos, suggests that at least what's in the images is available. So far the iso image has been downloading/creating for an hour, I installed this on a VM in Proxmox, and there is normal Ubuntu :) no IP addresses, ports from the graphics in the Readme work. That's why people are disappointed with this "marketing campaign" in recent days.

Thank you for understanding. It seems like the marketing folks have talked a lot, but that is indeed what we want to do. I will remove the os_installer folder before it can function properly.

The code for the Jarvis agent is executable(I have already used it to generate many images for me using Stable Diffusion.) To run the functions for analyzing YouTube videos and sending tweets, there is some configuration work that is a bit more complex. This is mainly because you need to obtain the relevant secret keys from Google and Twitter. I'm thinking of ways to simplify this process.

**Comment by Anon2578** [comment]
what happen to the jarvis core file can we really modify this project ourselves ?
i was trying to add configs into this project but it seems you all have it hard coded on what to say like for example it keeps mentioning opendan and cant even edit the configs to change that because the core file was deleted or is no longer available. Is this the purpose of this project ? Because i was going to add some login credentials to give to jarvis to use extend functionality but im not so sure since the core file was deleted im not sure if this project is really able to be modified and used how we want as it states. But i could be wrong can anyone help me with this issue because the jarvis core.py file seems to be gone or deleted and cant further add configs that suits me best.

**Comment by DiligentCatCat** [comment]
> what happen to the jarvis core file can we really modify this project ourselves ? i was trying to add configs into this project but it seems you all have it hard coded on what to say like for example it keeps mentioning opendan and cant even edit the configs to change that because the core file was deleted or is no longer available. Is this the purpose of this project ? Because i was going to add some login credentials to give to jarvis to use extend functionality but im not so sure since the core file was deleted im not sure if this project is really able to be modified and used how we want as it states. But i could be wrong can anyone help me with this issue because the jarvis core.py file seems to be gone or deleted and cant further add configs that suits me best.

What did you mean by 'core file'? Did you mean the file defining the basic information of Jarvis? If so, the gpt_agent.py might be what you want, and yes, this part is hard coded for now. You can find it in gpt_agent.py. And you can modify the prompt to whatever you want it to say.  
As for login credentials, I'm sorry to say that Jarvis does not support permission control for different sessions yet, it's now designed for private usage, in which case it's not necessary. If you want to add login credentials and provide different permissions for different sessions, it will take some effort to support. AFAICS, you will need to add authorization information into CallerContext (or session), and check permissions before invoking functions.

**Comment by Anon2578** [comment]
Thank you and can you help me add the llm models I want to Jarvis for
example I want to use some huggingface llm instead of openai but I’m not
100% sure how to add the file it link to the code. I see in the env it says
small llm and main llm I know adding the huggingface link or finding to
that don’t help do I have to download the file and then add it to the code
somehow I’m confused on that part and another thing with stable diffusion
the web address we’re do I get it from?

**Comment by DiligentCatCat** [comment]
No, you don't need to download any model. Jarvis is a pure network program, it does not do any prediction. All AI abilities are accessed through network request.

**Comment by Anon2578** [comment]
Thanks for the reply.
How do I specify the huggingface model I want Jarvis to use ?
Can you provide me an example on how to set this up ?
And what happens if I want to switch to a different model in the future?
And is their any resources or documentation you can recommend for learning
more about this because I seen the opendan videos with multiple models in
one OS system it was interesting and I been trying to do that for the
longest I’m excited to get started in this project.

**Comment by DiligentCatCat** [comment]
You can specify `JARVIS_OPENAI_URL_BASE` if the API is compatible with OpenAI API (but the output may be bad since the prompts are not tested).

If not, you'll have to write another agent, see `agent_jarvis/jarvis/ai_agent/webui_agent.py`, this is an unfinished code for `text-generation-webui`. You can write your own, and make `agent_facotry` return your own agent. You can also provide an OpenAI API 
for your own API, `text-generation-webui` has a built plugin for that.

Whatever the method you choose, you may need to take some effort to make it work. At least the prompts only suits ChatGPT now.

@Anon2578
